cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowDriver.java

```diff
@@ -178,6 +179,10 @@
                                                            workflowSpec.getLocalDatasetSpecs().keySet(),
                                                            workflowRunId.getRun());
     this.basicWorkflowToken = new BasicWorkflowToken(cConf.getInt(Constants.AppFabric.WORKFLOW_TOKEN_MAX_SIZE_MB));
+    if (workflowToken != null) {
+      this.basicWorkflowToken.mergeToken(workflowToken);
```

Comment:
The `workflowToken` contains just the USER scoped key-value pairs. Merging it was the easiest way of copying over all of the attributes.

We could also just do something like this though, if this is better:

```
this.basicWorkflowToken = (workflowToken != null) ? 
  workflowToken : 
  BasicWorkflowToken(cConf.getInt(Constants.AppFabric.WORKFLOW_TOKEN_MAX_SIZE_MB));